### PR TITLE
Vendor guidelines

### PR DESCRIPTION
**Pull request by CrystalOnScript**

### CODE REVIEW

File: pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+To qualify for inclusion, all contribution levels must:
+
+*   Meet the [notability requirements of the English Wikipedia](https://en.wikipedia.org/wiki/Wikipedia:Notability).
+*   Maintain or improve the same type of service level as AMP promises to both publishers and users. 
+*   Be created in good quality.
+*   Create a troubleshooting channel for their customers. 
+*   Provide good integration test coverage from their side to test against both AMP’s prod & canary releases.
+*   Fulfill a purpose that does not exist. 
+
+Vendor contributions are divided into three different levels. Each level is dependent on how much additional business logic and vendor specific logic (logic that is not shared with other vendors) is added to the AMP repo. A general principle is that the more logic, especially vendor specific logic, added to AMP, the higher contribution level is and the more commitment needed from the vendor.
```
Comment (sebastianbenz) [inline pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md]: > A general principle is that the more logic, especially vendor specific logic, added to AMP, the higher contribution level is and the more commitment needed from the vendor.

This is a bit hard to understand. Can you rephrase? Maybe split into two sentences.

File: pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+Level one & level two contributions share components between vendors. If there is a component fulfills a purpose similar to your business, consider reusing that component. This requires much less effort and is more long-term maintainable.
```
Comment (sebastianbenz) [inline pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md]: Hm. Have you considered using number literals for the different levels? `Level 1`, `Level 2`. These make it easier to parse the text.

File: pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+
+After deciding what level of contribution meets your use case, open a [GitHub issue](https://github.com/ampproject/amphtml/issues/new) to start. 
+
+
+## Level one contribution
+
+Level one contributions share components and load vendor specific logic as custom JavaScript in a cross origin iframe.
```
Comment (sebastianbenz) [inline pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md]: What does `share components` mean?

Comment (sebastianbenz) [reply pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md]: Maybe also add an example for a typical level one contribution in the first paragraph.

File: pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+After deciding what level of contribution meets your use case, open a [GitHub issue](https://github.com/ampproject/amphtml/issues/new) to start. 
+
+
+## Level one contribution
+
+Level one contributions share components and load vendor specific logic as custom JavaScript in a cross origin iframe. 
+
+Vendors add configurations or features to existing extension to implement their functionalities. If such a component does not exist they may propose a new one. 
+
+The only vendor specific logic checked into AMP repo is a vendor configuration. Adding a new vendor to an existing level one contribution typically does not need a design review. Vendors can follow the documentation of the component.
```
Comment (sebastianbenz) [inline pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md]: Can you elaborate on "follow the documentation of the component".

File: pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md (hunk 3 above)
Comment (sebastianbenz) [reply pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md]: I also don't understand the iframe part.

File: pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md [hunk 5]
```diff
@@ -0,0 +1,108 @@
...
+
+There are 90+ ad providers who have added support for advertising related features like amp-ad. Here is a [sample pull request](https://github.com/ampproject/amphtml/pull/2299) from the ad network [Criteo](https://github.com/ampproject/amphtml/blob/master/ads/criteo.md).
+
+## Level two contribution 
+
+Level two vendor contributions share components. All logic is checked into the AMP repository, and no custom Javascript can be loaded into an iframe.  Vendors add configurations or features to existing components to implement their functionalities. If such a component does not exist they may propose a new one.
```
Comment (sebastianbenz) [inline pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md]: Also: add an example in the first paragraph to make it more clear.

File: pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+
+There are 90+ ad providers who have added support for advertising related features like amp-ad. Here is a [sample pull request](https://github.com/ampproject/amphtml/pull/2299) from the ad network [Criteo](https://github.com/ampproject/amphtml/blob/master/ads/criteo.md).
+
+## Level two contribution 
+
+Level two vendor contributions share components. All logic is checked into the AMP repository, and no custom Javascript can be loaded into an iframe.  Vendors add configurations or features to existing components to implement their functionalities. If such a component does not exist they may propose a new one. 
+
+All business logic is checked into AMP repo, but the only vendor specific logic checked in is a vendor configuration. If component works with a vendor provided config file, no design review is needed. If the vendor configuration implements a new feature or new component it will need to pass AMP’s design review.
```
Comment (sebastianbenz) [inline pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md]: Add an example for business logic and vendor specific logic to make it more clear what these mean.

File: pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md (hunk 2 above)
Comment (CrystalOnScript) [reply pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md]: That's the journalist in me - trained to write out numbers until 10. I'll update it

File: pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md (hunk 3 above)
Comment (lannka) [reply pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md]: > I also don't understand the iframe part.

Those level one vendor integrations leverage a cross-origin iframe to run vendor specific logic as custom javascript. Those javascript are not checked into AMP repo.

Comment (lannka) [reply pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md]: > What does share components mean?

Maybe `share extensions` is more specific in AMP's context. e.g., many ad networks share the `amp-ad` extension.

**Comment by CrystalOnScript** [comment]
@sebastianbenz - PTAL, I think its shaping up to be a bit better

File: pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+
+There are many ad providers who have added support for advertising related features like amp-ad. Here is a [sample pull request](https://github.com/ampproject/amphtml/pull/2299) from the ad network [Criteo](https://github.com/ampproject/amphtml/blob/master/ads/criteo.md).
+
+## Level 2 contribution 
+
+Level 2 contributions leverage the feature logic of existing components. All logic is checked into the AMP repository, and no custom Javascript can be loaded into an iframe. For example, analytics providers add their configurations to the [`amp-analytics`](../../../components/reference/amp-analytics.md) component but including the endpoint to track data, such as user clicks.  
+
+Vendors add configurations or features, such as new APIs, to existing components to implement their functionalities. If such a component does not exist they may propose a new one. 
+
+All business logic is checked into AMP repository, but the only vendor specific logic checked in is a vendor configuration. If component works with a vendor provided config file, no design review is needed. If the vendor configuration implements a new feature or new component it will need to pass AMP’s design review.
```
Comment (sebastianbenz) [inline pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md]: nit: `If component works` maybe `If the component works`?

File: pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+Level 2 contributions leverage the feature logic of existing components. All logic is checked into the AMP repository, and no custom Javascript can be loaded into an iframe. For example, analytics providers add their configurations to the [`amp-analytics`](../../../components/reference/amp-analytics.md) component but including the endpoint to track data, such as user clicks.  
+
+Vendors add configurations or features, such as new APIs, to existing components to implement their functionalities. If such a component does not exist they may propose a new one. 
+
+All business logic is checked into AMP repository, but the only vendor specific logic checked in is a vendor configuration. If component works with a vendor provided config file, no design review is needed. If the vendor configuration implements a new feature or new component it will need to pass AMP’s design review. 
+
+### Vendor expectations
+
+*   Adding a new vendor to an existing level 2 contribution typically does not need a design review. The vendor can follow the documentation of that component. 
+*   Proposing a new component for level 2 contribution will need to be generic and shareable by other vendors.
```
Comment (sebastianbenz) [inline pages/content/amp-dev/documentation/guides-and-tutorials/contribute/vendor-contributions/index.md]: This sentence is not clear to me.
